src/VisualStudio/Core/Def/Implementation/LanguageClient/InProcLanguageServer.cs

```diff
@@ -0,0 +1,232 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.FindUsages;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.ErrorReporting;
+using Microsoft.CodeAnalysis.LanguageServer;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Newtonsoft.Json.Linq;
+using StreamJsonRpc;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.LanguageService
+{
+    internal class InProcLanguageServer
+    {
+        private readonly IDiagnosticService _diagnosticService;
+        private readonly JsonRpc _jsonRpc;
+        private readonly IThreadingContext _threadingContext;
+        private readonly LanguageServerProtocol _protocol;
+        private readonly Workspace _workspace;
+
+        private VSClientCapabilities? _clientCapabilities;
+
+        public InProcLanguageServer(Stream inputStream, Stream outputStream, LanguageServerProtocol protocol, Workspace workspace, IDiagnosticService diagnosticService, IThreadingContext threadingContext)
+        {
+            this._protocol = protocol;
+            this._workspace = workspace;
+            this._threadingContext = threadingContext;
+
+            this._jsonRpc = new JsonRpc(outputStream, inputStream, this);
+            this._jsonRpc.StartListening();
```

Comment:
any chance these throw/fail?

Reply:
yes, it is possible, but that would mean we're in a very very broken state (e.g. the stream was created with the wrong properties / _jsonRpc is disposed).  Either way, liveshare / VS lsp client handles displaying and recording the error.  I don't know that there is anything we can do to recover here.